keps/sig-node/3857-rro-hostpath/README.md

```diff
@@ -0,0 +1,1046 @@
+
+### Non-Goals
+
+<!--
+What is out of scope for this KEP? Listing non-goals helps to focus discussion
+and make progress.
+-->
+Support recursively read-only mounts for old runc and old kernel releases.
+
+## Proposal
+
+<!--
+This is where we get down to the specifics of what the proposal actually is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation. What is the desired outcome and how do we measure success?.
+The "Design Details" section below is for the real
+nitty-gritty.
+-->
+
+### User Stories (Optional)
+
+<!--
+Detail the things that people will be able to do if this KEP is implemented.
+Include as much detail as possible so that people can understand the "how" of
+the system. The goal here is to make this feel real for users without getting
+bogged down.
+-->
+
+#### Story 1
+
+A user wants to mount `/mnt`, includings its submounts such as `/mnt/usbstorage`, as read-only.
+
+### Notes/Constraints/Caveats (Optional)
+
+<!--
+What are the caveats to the proposal?
+What are some important details that didn't come across above?
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+Constraints: needs runc >= 1.1 && kernel >= 5.12.
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal, and how do we mitigate? Think broadly.
+For example, consider both security and how this will impact the larger
+Kubernetes ecosystem.
+
+How will security be reviewed, and by whom?
+
+How will UX be reviewed, and by whom?
+
+Consider including folks who also work outside the SIG or subproject.
+-->
+
+Risk: a manifest with `recursivelyReadOnly: Enabled` will not work on old distros (See "Constraints" above).
+
+Mitigation: set `recursivelyReadOnly: BestEffort`.
+
+## Design Details
+
+<!--
+This section should contain enough information that the specifics of your
+change are understandable. This may include API specs (though not always
+required) or even code snippets. If there's any ambiguity about HOW your
+proposal will be implemented, this is the place to discuss them.
+-->
+
+
+### Core API
+Add `RecursivelyReadOnly: (Disabled|BestEffort|Enabled)` to the [`VolumeMount`](https://github.com/kubernetes/kubernetes/blob/v1.26.1/pkg/apis/core/types.go#L1854-L1880) struct:
+- `Disabled` (default): always non-recursive
+- `BestEffort`: recursive if possible, otherwise non-recursive, without raising an error.
+   This value is translated to `Enabled` or `Disabled` in the `VolumeMount` struct (see the "Diff" below).
+- `Enabled`: recursive if possible, otherwise raise an error (if the node components are aware of this feature).
+
+When `RecursivelyReadOnly` is set to any non-nil value, `ReadOnly` is automatically implied to be `true`.
+When `RecursivelyReadOnly` is set to `BestEffort` or `Enabled`, `MountPropagation` should be set to `"None"` (default).
```

Comment:
What does "should" mean?  Does it mean thast is the user's problem?  Or does it mean we change the default?  Or does it mean something else?

Reply:
The user's problem.

Reworded to:
```
When a user sets `RecursivelyReadOnly` to `BestEffort` or `Enabled`, they should also set `MountPropagation` to `"None"` (default).
Otherwise, a writable submount may be propagated from the host mount namespace.
```